### PR TITLE
Added nginx-sm entry

### LINKED ISSUE #542: Add NGINX Service Mesh results to the SMI Conformance Dashboard
**Current State of the Meshery.io site**
https://meshery.io/smi is missing an entry for NGINX Service Mesh.

**Desired State of the Meshery.io site**
Make the same results that are available here - https://layer5.io/service-mesh-landscape#smi, available on meshery.io, too.

---
### Contributor Resources

The meshery.io website uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/meshery/meshery.io/tree/master).
- See the [Contributing to Meshery.io Website](https://github.com/layer5io/meshery.io#contributing-to-the-mesheryio-website) section of the readme.md and other [contributing instructions](https://docs.meshery.io/project/contributing), too.
- See Meshery site designs in this [Figma designfile](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI?node-id=110%3A1). Fill-in the [Layer5 Community Member Form](https://layer5.io/newcomer) and join the [Layer5 Community Slack](http://slack.layer5.io) for access.

### PR DESCRIPTION
**Pull request by afzal442**
Signed-off-by: afzal442 <afzal442@gmail.com>

**Description**

This PR fixes #542 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@dhruv0000 please review today.

**Comment by leecalcote** [comment]
@afzal442 I see the `pr/draft` label. Is the PR ready for review or not?

File: collections/_pages/smi.html [hunk 1]
```diff
@@ -45,13 +45,14 @@
         img.setAttribute('title', result.mesh_name)
 
         switch (result.mesh_name) {
-          case "Open Service Mesh": img.setAttribute('src', 'images/osm.svg'); break;
           case "Traefik Mesh": img.setAttribute('src', 'images/traefik-mesh.svg'); break;
-          case "Istio": img.setAttribute('src', 'images/istio.svg'); break;
+          case "Open Service Mesh": img.setAttribute('src', 'images/osm.svg'); break;
+          case "Nginx": img.setAttribute('src', 'images/nginx-sm.svg'); break;
```
Comment (dhruv0000) [inline collections/_pages/smi.html:50]: Use `Nginx Service Mesh` here.

The names are basically parsed version of the smp spec name (if you want to refer in the future) https://github.com/service-mesh-performance/service-mesh-performance/blob/master/spec/service_mesh.pb.go

**Review by dhruv0000 — CHANGES_REQUESTED** [review]

File: collections/_pages/smi.html (hunk 1 above)
Comment (afzal442) [reply collections/_pages/smi.html:50]: https://meshery.layer5.io/smi/results/public; it is found out here also mainly; thanks;

File: collections/_pages/smi.html [hunk 2]
```diff
@@ -45,13 +45,14 @@
...
           case "Linkerd": img.setAttribute('src', 'images/linkerd.svg'); break;
           case "Kuma": img.setAttribute('src', 'images/Kuma.svg'); break;
-          case "Consul": img.setAttribute('src', 'images/consul.svg'); break;
-          case "App Mesh": img.setAttribute('src', 'images/aws-app-mesh.svg'); break;
+          // case "Consul": img.setAttribute('src', 'images/consul.svg'); break;
+          // case "App Mesh": img.setAttribute('src', 'images/aws-app-mesh.svg'); break;
```
[the remark this answers is not in the document]
Comment (afzal442) [reply collections/_pages/smi.html:54]: ok reverting!

File: _data/smi.json [hunk 3]
```diff
@@ -1,312 +1,224 @@
 [
     {
-        "id": "0dd368be-745f-4433-a356-f717aac40457",
```
[the remark this answers is not in the document]
Comment (afzal442) [reply _data/smi.json:3]: thanks, alr8, deleting.

**Comment by afzal442** [comment]
hi @AnshumanDhiman, I have spent long hrs debugging this. Could you look into it? Need your kind help. Thanks.

**Comment by leecalcote** [comment]
> hi @AnshumanDhiman, I have spent long hrs debugging this. Could you look into it? Need your kind help. Thanks.

What have you already tried? What isn't working?

**Comment by afzal442** [comment]
sorry @leecalcote! what I have found so far and debugged that you can see on the screen shot;
c2
c3
so here `mesh_version` is taken as none and currently this `mesh_name` is getting repeated;

HTMS!

**Comment by dhruv0000** [comment]
> HTMS

@afzal442 So mesh name getting repeated is fine, till they are of different version, if they are of same version, the latest one is selected. And I think you found the error yourself. The NGINX test had no version while running the test (i.e. mesh_version = none). Ig I will run another test with the proper version setup and this should be fixed.

**Comment by afzal442** [comment]
@dhruv0000 TSM!

> if service meshes are of same version, the latest one is selected.

How to select the latest one? Or can we add the latest ones on smi.json file where we can allow to fetch the latest ones?

> The NGINX test had no version while running the test

I can go like this if(mesh_version === none && mesh_name !== 'Nginx Service Mesh') to fetch 'Nginx SM'

> I will run another test with the proper version setup and this should be fixed.

Sure!

**Comment by warunicorn19** [comment]
> > HTMS
> 
> @afzal442 So mesh name getting repeated is fine, till they are of different version, if they are of same version, the latest one is selected. And I think you found the error yourself. The NGINX test had no version while running the test (i.e. mesh_version = none). Ig I will run another test with the proper version setup and this should be fixed.

@dhruv0000 did you get the chance to run the test?

**Comment by afzal442** [comment]
@leecalcote you can see the recent commit makes it possible to fetch the latest ones of smi compliances; have a look at screenshot as well;
c4


Once the test is done with the proper version setup, Nginx will come into this place.

**Comment by leecalcote** [comment]
@afzal442 please engage with @dhruv0000

**Comment by warunicorn19** [comment]
@leecalcote, @dhruv0000 has already run the test, currently in meshery.io there's no logo for NGINX because we haven't put up one yet, but the data is now there. I think we can merge this PR.

**Comment by dhruv0000** [comment]
Looks good. Lets merge.

> @leecalcote, @dhruv0000 has already run the test, currently in meshery.io there's no logo for NGINX because we haven't put up one yet, but the data is now there. I think we can merge this PR.

Ya. the test is up. I think we are good to merge,

**Review by dhruv0000 — APPROVED** [review]
Looks good. Lets merge.

**Comment by afzal442** [comment]
image
LGTM!

**Pull request merged**